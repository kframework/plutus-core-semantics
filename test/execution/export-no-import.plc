(program

(module Foo (import) (export () (r1))
  (declare r1 (integer))
  (define r1 2)
  (define r2 (builtin equalsInteger 2 3))
  (define r5 (builtin equalsByteString #AAA #AAA))
  (define r6 (builtin equalsByteString #AAA #111))
)

(module Bar (import) (export () ())
  (define r7 (builtin equalsInteger Foo.r1 2))
)

)
