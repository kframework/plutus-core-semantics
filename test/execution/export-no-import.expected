<T>
  <k>
    Foo . r1 ~> #freezer(builtin___)_PLUTUS-CORE-COMMON1_ ( equalsInteger , 2 ) ~> #freezer(define__)_PLUTUS-CORE-COMMON0_ ( r7 ) ~> .DecList ~> ( program .ModList )
  </k>
  <lenv>
    .Map
  </lenv>
  <genv>
    Foo . r1 |-> 2
    Foo . r2 |-> ( con Prelude . False .ValList )
    Foo . r5 |-> ( con Prelude . True .ValList )
    Foo . r6 |-> ( con Prelude . False .ValList )
  </genv>
  <modules>
    ModuleCellSetItem ( <Module>
      <curr>
        0
      </curr>
      <name>
        Foo
      </name>
      <dataCons>
        .TyExpList
      </dataCons>
      <decl>
        r1  .QualCList
      </decl>
      <importedModules>
        .ModNmList
      </importedModules>
      <exportTy>
        .TyExpList
      </exportTy>
      <exportTm>
        r1  .QualCList
      </exportTm>
    </Module> ) ModuleCellSetItem ( <Module>
      <curr>
        1
      </curr>
      <name>
        Bar
      </name>
      <dataCons>
        .TyExpList
      </dataCons>
      <decl>
        .QualCList
      </decl>
      <importedModules>
        .ModNmList
      </importedModules>
      <exportTy>
        .TyExpList
      </exportTy>
      <exportTm>
        .QualCList
      </exportTm>
    </Module> )
  </modules>
</T>
